library/ssl_tls.c

```diff
@@ -4224,14 +4224,26 @@ int mbedtls_ssl_config_defaults( mbedtls_ssl_config *conf,
             conf->min_major_ver = MBEDTLS_SSL_MIN_MAJOR_VERSION;
             conf->max_major_ver = MBEDTLS_SSL_MAX_MAJOR_VERSION;
 
-            conf->min_minor_ver = MBEDTLS_SSL_MINOR_VERSION_3; /* TLS 1.2 */
-#if defined(MBEDTLS_SSL_PROTO_TLS1_2) && defined(MBEDTLS_SSL_PROTO_TLS1_3)
-            /* Hybrid TLS 1.2/1.3 is not supported yet */
-            conf->max_minor_ver = MBEDTLS_SSL_MINOR_VERSION_3;
+            if( ( endpoint == MBEDTLS_SSL_IS_SERVER ) ||
+                ( transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM ) )
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2)
+            {
+                conf->min_minor_ver = MBEDTLS_SSL_MINOR_VERSION_3;
+                conf->max_minor_ver = MBEDTLS_SSL_MINOR_VERSION_3;
+            }
 #else
-            conf->max_minor_ver = MBEDTLS_SSL_MAX_MINOR_VERSION;
-#endif /* MBEDTLS_SSL_PROTO_TLS1_2 && MBEDTLS_SSL_PROTO_TLS1_3 */
-
+            {
+                conf->min_major_ver = 0;
+                conf->max_major_ver = 0;
+                conf->min_minor_ver = 0;
+                conf->max_minor_ver = 0;
+            }
+#endif
+            else
+            {
+                conf->min_minor_ver = MBEDTLS_SSL_MIN_MINOR_VERSION;
+                conf->max_minor_ver = MBEDTLS_SSL_MAX_MINOR_VERSION;
+            }
             conf->ciphersuite_list = ssl_preset_suiteb_ciphersuites;
```

Comment:
Note: in practice this restricts to 1.2 as this only includes 1.2 ciphersuites.

Reply:
That's right thus not coherent with the setting of the maximum version to TLS 1.3. Thus I guess we should only go with TLS 1.2 is that case, right ?

Reply:
I think so, yes. I think if we offer TLS 1.3 with no 1.3 ciphersuites, the server is likely to first pick a version, then try to pick a ciphersuite and fail, then fail the handshake, which it not super helpful. Also the abstract of [RFC 6460](https://datatracker.ietf.org/doc/html/rfc6460) says "This document defines a profile of
   Transport Layer Security (TLS) version 1.2" - though it was written before TLS 1.3 was a thing.